README.md

```diff
@@ -3,16 +3,133 @@ Dagger Reflect
+
+```gradle
+dependencies {
+  implementation "com.google.dagger:dagger:${VERSION_DAGGER}"
+  if (project.hasProperty('android.injected.invoked.from.ide')) {
+    implementation "com.jakewharton.dagger:dagger-reflect:${VERSION_DAGGER_REFLECT}"
+    annotationProcessor "com.jakewharton.dagger:dagger-reflect-compiler:${VERSION_DAGGER_REFLECT}"
+  } else {
+    annotationProcessor "com.google.dagger:dagger-compiler:${VERSION_DAGGER}"
+  }
+}
+```
+
+_(No need to upgrade Proguard shrinking configuration with this approach.)_
+
+### Pure reflection, no APT
+This usage type replaces everything with reflection, but requires a few trivial code changes across the app.
+
+```gradle
+dependencies {
+  implementation "com.google.dagger:dagger:${VERSION_DAGGER}"
+  if (project.hasProperty('android.injected.invoked.from.ide')) {
+    implementation "com.jakewharton.dagger:dagger-reflect:${VERSION_DAGGER_REFLECT}"
+  } else {
+    implementation "com.jakewharton.dagger:dagger-codegen:${VERSION_DAGGER_REFLECT}"
+    annotationProcessor "com.google.dagger:dagger-compiler:${VERSION_DAGGER}"
+  }
+}
+```
+
+To bridge the compatibility between `dagger-compile` generated classes and `dagger-reflect`, you'll have to use `dagger.Dagger` class (from `dagger-codegen` or `dagger-reflect`) as a replacement:
+ * `FooComponent foo = DaggerFooComponent.create()`  
+   &rarr; `FooComponent foo = Dagger.create(FooComponent.class)`
+ * `FooComponent foo = DaggerFooComponent.builder().….build()`  
+   &rarr; `FooComponent foo = Dagger.builder(FooComponent.Builder.class).….build()`
+
+For the release build using `dagger-codegen` and shirking you'll need to add this to the configuration:
+```proguard
+# [dagger-reflect] Make sure to keep entry points that dagger.Dagger reflectively accesses
+# fixes: Unable to find generated component implementation com.example.Daggera for component com.example.a
+
+# Keep annotation to be able to match it below
+-keep class dagger.Component
+# Keep the names of the Component interfaces so we can prefix them with "Dagger"
+-keepnames @dagger.Component interface **
+# Keep the names of the generated Component implementations so the prefixed lookup succeeds
+-keepnames class ** implements @dagger.Component **
+# Keep the builder() method in the generated Component to reflectively call it
+-keepclassmembers class ** implements @dagger.Component ** {
+	public static ** builder();
+}
+# Keep the create() method in the generated Component to reflectively call it
+-keepclassmembers class ** implements @dagger.Component ** {
+	public static <2> create();
+}
+```
```

Comment:
These rules can be embedded in the jar to be picked up automatically and we can add test cases to ensure they work. They don't need to be documented anywhere.